discoverer/src/main/java/eu/seaclouds/platform/discoverer/core/Offering.java

```diff
@@ -171,12 +172,17 @@ public String toTosca() {
         if (this.toscaString != null)
             return this.toscaString;
 
-        /* otherwise, if there is the path of the file where the offering is stored, the file is read */
-        if (this.offeringPath != null) {
+        /* otherwise, if the offering is stored in the repository, it is possible to read it from file */
+        if (this.isStored) {
             try {
-                this.toscaString = new String(Files.readAllBytes(Paths.get(this.offeringPath)));
+                String offeringDirectory = Discoverer.instance().offeringManager.getOfferingDirectory().getAbsolutePath();
+                String offeringFileName = "offer_" + this.offeringId + ".yaml";
+                this.toscaString = new String(Files.readAllBytes(Paths.get(offeringDirectory + File.separatorChar + offeringFileName)));
                 return this.toscaString;
-            } catch (IOException e) { }
+            } catch (IOException e) {
+                /* Cannot read the offering from repository */
+                e.printStackTrace();
```

Comment:
You should be using log api. In this case, log.error(). The same applies to all e.printStackTrace or System.err.print throughout the code.